[2025-09-07T17:44:35.778+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ml_pipeline.model_training manual__2025-09-07T17:44:14.008911+00:00 [queued]>
[2025-09-07T17:44:35.792+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ml_pipeline.model_training manual__2025-09-07T17:44:14.008911+00:00 [queued]>
[2025-09-07T17:44:35.793+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-09-07T17:44:35.814+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): model_training> on 2025-09-07 17:44:14.008911+00:00
[2025-09-07T17:44:35.823+0000] {standard_task_runner.py:60} INFO - Started process 4642 to run task
[2025-09-07T17:44:35.831+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'ml_pipeline', 'model_training', 'manual__2025-09-07T17:44:14.008911+00:00', '--job-id', '215', '--raw', '--subdir', 'DAGS_FOLDER/ml_pipeline.py', '--cfg-path', '/tmp/tmpmi181n3g']
[2025-09-07T17:44:35.834+0000] {standard_task_runner.py:88} INFO - Job 215: Subtask model_training
[2025-09-07T17:44:35.943+0000] {task_command.py:423} INFO - Running <TaskInstance: ml_pipeline.model_training manual__2025-09-07T17:44:14.008911+00:00 [running]> on host c1a35c3f3bc1
[2025-09-07T17:44:36.130+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='sarah' AIRFLOW_CTX_DAG_ID='ml_pipeline' AIRFLOW_CTX_TASK_ID='model_training' AIRFLOW_CTX_EXECUTION_DATE='2025-09-07T17:44:14.008911+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-09-07T17:44:14.008911+00:00'
[2025-09-07T17:45:20.895+0000] {logging_mixin.py:188} INFO - STDOUT:
[2025-09-07T17:45:20.898+0000] {logging_mixin.py:188} INFO -  Fitting 3 folds for each of 10 candidates, totalling 30 fits
üèÉ View run XGBoost_RandSearch at: https://dagshub.com/SamMebarek/mlops-som.mlflow/#/experiments/0/runs/9b00b02677a247ffb01e164677f79c88
üß™ View experiment at: https://dagshub.com/SamMebarek/mlops-som.mlflow/#/experiments/0
[2025-09-07T17:45:20.899+0000] {logging_mixin.py:188} INFO - STDERR:
[2025-09-07T17:45:20.900+0000] {logging_mixin.py:188} INFO -  2025-09-07 17:44:42,702 - training.utils.common - INFO - YAML charg√© : /opt/***/project/config/config.yaml
2025-09-07 17:44:42,710 - training.utils.common - INFO - YAML charg√© : /opt/***/project/config/params.yaml
2025-09-07 17:44:42,712 - training.utils.common - INFO - R√©pertoire cr√©√© ou existant : /opt/***/project/models
2025-09-07 17:44:42,712 - training.utils.common - INFO - R√©pertoire cr√©√© ou existant : /opt/***/project/models
2025-09-07 17:44:43,672 - __main__ - INFO - Loaded processed data: /opt/***/project/data/processed/clean_data.csv (shape=(108060, 14))
2025-09-07 17:44:44,371 - __main__ - INFO - Split data: train_shape=(86448, 11), test_shape=(21612, 11)
2025/09/07 17:44:44 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

2025-09-07 17:44:44,778 - __main__ - INFO - Starting hyperparameter search and training.
2025-09-07 17:45:01,331 - __main__ - INFO - Best params: {'colsample_bytree': 0.9219644132807024, 'gamma': 0.06692515264078093, 'learning_rate': 0.19307895078298942, 'max_depth': 5, 'n_estimators': 86, 'subsample': 0.6271914913054505}
2025-09-07 17:45:01,331 - __main__ - INFO - R^2 score: 0.9998
2025/09/07 17:45:18 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: PriceFixerModel, version 99
2025-09-07 17:45:18,592 - __main__ - INFO - Registered model 'PriceFixerModel' from source=mlflow-artifacts:/4829b4e5ce254fea8f678e40a4133671/9b00b02677a247ffb01e164677f79c88/artifacts/xgb_model, run_id=9b00b02677a247ffb01e164677f79c88
2025-09-07 17:45:19,039 - training.repository.repository - INFO - Mod√®le sauvegard√© ‚Üí /opt/***/project/models/xgb_model.pkl
2025-09-07 17:45:19,070 - __main__ - INFO - Local MLflow dir: models/xgb_model_mlflow_manual__2025-09-07T17:44:14.008911+00:00 (size=244004 bytes)
2025-09-07 17:45:19,071 - __main__ - INFO - Model saved locally at: /opt/***/project/models/xgb_model.pkl
2025-09-07 17:45:19,074 - __main__ - INFO - Pushed metrics to Pushgateway (http://pushgateway:9091) status=1 duration_s=36.391 rows=108060 r2=0.9997696380635521 artifact_bytes=484994
[2025-09-07T17:45:20.901+0000] {python.py:201} INFO - Done. Returned value was: None
[2025-09-07T17:45:20.916+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=ml_pipeline, task_id=model_training, execution_date=20250907T174414, start_date=20250907T174435, end_date=20250907T174520
[2025-09-07T17:45:20.960+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2025-09-07T17:45:21.007+0000] {taskinstance.py:3280} INFO - 1 downstream tasks scheduled from follow-on schedule check
